pkg/controllers/nodeclass/controller.go

```diff
@@ -222,6 +232,39 @@ func (c *Controller) resolveInstanceProfile(ctx context.Context, nodeClass *v1be
 	return nil
 }
 
+func (c *Controller) updateNodeClaimHash(ctx context.Context, nodeClass *v1beta1.EC2NodeClass) error {
+	ncList := &corev1beta1.NodeClaimList{}
+	if err := c.kubeClient.List(ctx, ncList, client.MatchingFields{"spec.nodeClassRef.name": nodeClass.Name}); err != nil {
+		return err
+	}
+
+	for i := range ncList.Items {
+		stored := ncList.Items[i].DeepCopy()
+
+		if ncList.Items[i].Annotations[v1beta1.EC2NodeClassHashVersion] != v1beta1.EC2NodeClassHashVersion {
+			ncList.Items[i].Annotations = lo.Assign(ncList.Items[i].Annotations, map[string]string{
+				v1beta1.AnnotationEC2NodeClassHashVersion: v1beta1.EC2NodeClassHashVersion,
+			})
+
+			// Any NodeClaim that is already drifted will remain drifted if the karpenter.sh/ec2nodeclass-hash-version doesn't match
+			// Since the hashing mechanism has changed we will not be able to determine if the drifted status of the node has changed
+			if ncList.Items[i].StatusConditions().GetCondition(corev1beta1.Drifted) == nil {
+				ncList.Items[i].Annotations = lo.Assign(ncList.Items[i].Annotations, map[string]string{
+					v1beta1.AnnotationEC2NodeClassHash: nodeClass.Hash(),
+				})
+			}
```

Comment:
Have the same comment here as in the kubernetes-sigs. We probably want to not update the hash version if it's drifted as well

Reply:
We will not update the nodepool-hash on the NodeClaim, if the nodepool-hash-version match the NodePool